src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpNamingStyleDiagnosticAnalyzer.cs

```diff
@@ -8,9 +10,27 @@ namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.NamingStyles
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     internal sealed class CSharpNamingStyleDiagnosticAnalyzer : NamingStyleDiagnosticAnalyzerBase
     {
-        public CSharpNamingStyleDiagnosticAnalyzer()
+        protected sealed override void OnCompilationStartAction(
+            CompilationStartAnalysisContext context,
+            ConcurrentDictionary<Guid, ConcurrentDictionary<string, string>> idToCachedResult)
         {
+            // HACK: RegisterSymbolAction doesn't work with local functions
+            context.RegisterSyntaxNodeAction(c => SyntaxNodeAction(c, idToCachedResult), SyntaxKind.LocalFunctionStatement);
+        }
+
+        private void SyntaxNodeAction(
+            SyntaxNodeAnalysisContext context,
+            ConcurrentDictionary<Guid, ConcurrentDictionary<string, string>> idToCachedResult)
+        {
+            var symbolContext = new SymbolAnalysisContext(
+                context.SemanticModel.GetDeclaredSymbol(context.Node, context.CancellationToken),
+                context.Compilation,
+                context.Options,
+                context.ReportDiagnostic,
+                isSupportedDiagnostic: _ => true,
+                context.CancellationToken);
 
+            SymbolAction(symbolContext, idToCachedResult);
```

Comment:
instead of this approach, i woudl prefer if we registered a Symbol-op and a Syntax-op, and they just called through to the same final helper method, rather than synthesizing up a SymbolAnalysisContext instead and having the Syntax-op call through the Symbol-op version.

Reply:
I don't know how. They use ReportDiagnostic from the context, Options and there is also an extension method on it that SymbolAction uses.

Reply:
fixed